core/src/poh_service.rs

```diff
@@ -47,9 +47,14 @@ impl PohService {
                     // Let's dedicate one of the CPU cores to this thread so that it can gain
                     // from cache performance.
                     if let Some(cores) = core_affinity::get_core_ids() {
-                        core_affinity::set_for_current(cores[0]);
+                        core_affinity::set_for_current(cores[cores.len() - 1]);
```

Comment:
Is there a strategy here?  (Maybe avoid being in the scheduler's way?)

I think we've mostly trained the community to expect core 0 to be 100% all of the time.  This change will probably bring some confusion, especially to those who've been toying with throttling via sysfs knobs.

NP if there's a good reason, but we should probably note the change in release notes and be ready for questions

Reply:
This is my thinking:
https://unix.stackexchange.com/questions/301471/can-the-0th-physical-core-used-asymmetrically-on-linux
https://serverfault.com/questions/446173/cpu0-is-swamped-with-eth1-interrupts

Basically core 0 is special on most systems in that the kernel by-default handles extra work there, so I think core 0 is not a good core to pin continuous work to.

Yes, we can add it to the release notes, I don't think the throttling is wide-spread and also it is very trivial to change the throttling to a different core. It's just a single number.

Reply:
FWIW, we have moved the POH service from core 0 to core 1 on our servers for the reason noted above.

Reply:
> This is my thinking:
> https://unix.stackexchange.com/questions/301471/can-the-0th-physical-core-used-asymmetrically-on-linux
> https://serverfault.com/questions/446173/cpu0-is-swamped-with-eth1-interrupts
> 
> Basically core 0 is special on most systems in that the kernel by-default handles extra work there, so I think core 0 is not a good core to pin continuous work to.

Oh, nice find!  A comment there would be great.

Perusing `/proc/interrupts` and  `/proc/irq/*/` on a couple systems, it looks like all cores are available in _some_ IRQ's affinity list and _most_ cores are assigned at least one IRQ.  The GCP validators have serviced more interrupts on `N-1` than `0` due to `virtio0-request` (I haven't figured out what that is yet).  My 3900x workstation has assigned an NVMe IRQ to core `N-1`.

I bring these observations up as a "further research may prove fruitful" rather than any sort of contention.  The change seems worth a try in any case

> Yes, we can add it to the release notes, I don't think the throttling is wide-spread and also it is very trivial to change the throttling to a different core. It's just a single number.

Yeah for sure.  There won't be any real problems.  I just expect a few torches and pitchforks :wink:

Reply:
Hmm, that is interesting. I can back this out until further testing. I expect the difference to be subtle and I didn't notice anything particularly different in my testing.

Reply:
That might not be a bad idea. I came across a study using `virtio0-request` as a proxy for virtualization environment efficiency.  All cores are in its affinity list though, so presumably it'd migrate.

If you want to back it out here, I can follow up with a hidden flag so we can experiment more easily